dvc/repo/checkout.py

```diff
@@ -12,6 +13,31 @@ def _cleanup_unused_links(self, all_stages):
     self.state.remove_unused_links(used)
 
 
+class ProgressCallback(object):
+    def __init__(self, total):
+        self.total = total
+        self.current = 0
+
+    def update(self, name, progress_to_add=1):
+        self.current += progress_to_add
+        progress.update_target(name, self.current, self.total)
+
+    def finish(self, name):
+        progress.finish_target(name)
+
+
+def get_all_files_numbers(stages):
+    return sum(stage.get_all_files_number() for stage in stages)
+
+
+def get_progress_callback(stages):
+    try:
+        total_files_num = get_all_files_numbers(stages)
+        return ProgressCallback(total_files_num)
+    except Exception:
```

Comment:
Not sure why we need this.

Reply:
There is a case (test_checkout.TestCheckoutMissingMd5InStageFile) when we miss md5 in stage file. On such event we will be trying to get files number for "None". Checkout will fail on `remote.local.get_files_number` on `self.is_dir_cache` check. I assumed that in that case (at least for now) it is better to just skip progress bar. Do you think We should write logic for obtaining file number from remote cache?

Reply:
@pared So why not make those functions not raise an exception or at least raise something specific and not shotgun everything here with `except Exception` that is way to general for this application? :slightly_smiling_face:

Reply:
Well, that is a good point.